### PR TITLE
Add variable `USE_LOCAL_KIND_REGISTRY` to CAPZ presubmit job

### PR DESCRIPTION
**Pull request by Jont828**
Add variable `USE_LOCAL_KIND_REGISTRY` based on this [CAPZ PR](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2823)

### CODE REVIEW

**Comment by dims** [comment]
please add comments in the PR body with context and what you are trying to fix

**Comment by Jont828** [comment]
For sure, I was just waiting on another PR to get ready before merging stuff this one.

**Comment by Jont828** [comment]
@CecileRobertMichon

**Comment by CecileRobertMichon** [comment]
/hold

need to set both variables side by side as discussed offline

File: config/prow/config.yaml [hunk 1]
```diff
@@ -937,6 +937,8 @@ presets:
       value: capzci.azurecr.io
     - name: LOCAL_ONLY
       value: "false"
+    - name: USE_LOCAL_KIND_REGISTRY
```
Comment (CecileRobertMichon) [inline config/prow/config.yaml:940]: can you also add a comment to `LOCAL_ONLY` that it is there for backward compatibility reasons and should be removed once CAPZ v1.7.0+ is used in all jobs?

**Comment by Jont828** [comment]
@CecileRobertMichon Done, I think we're good to go!

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Review by CecileRobertMichon — COMMENTED** [review]
/hold cancel

**Pull request merged**